### PR TITLE
Weather api

### PR DESCRIPTION
**Pull request by MixonZ**

### CODE REVIEW

File: src/main/java/io/codelex/flightplanner/api/TripWeather.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package io.codelex.flightplanner.api;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.time.LocalDateTime;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class TripWeather extends Trip {
```
Comment (sandris-) [inline src/main/java/io/codelex/flightplanner/api/TripWeather.java]: is this created from json?

File: src/main/java/io/codelex/flightplanner/api/Weather.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package io.codelex.flightplanner.api;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class Weather {
```
Comment (sandris-) [inline src/main/java/io/codelex/flightplanner/api/Weather.java]: is this created from json?

File: src/main/java/io/codelex/flightplanner/api/Weather.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    public Double getTemp_c() {
+        return temperature;
+    }
+
+    public void setTemp_c(Double temperature) {
+        this.temperature = temperature;
+    }
+
+    public Double getPrecip_mm() {
```
Comment (sandris-) [inline src/main/java/io/codelex/flightplanner/api/Weather.java]: what are these methods for?

File: src/main/java/io/codelex/flightplanner/weather/CacheKey.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package io.codelex.flightplanner.weather;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+class CacheKey {
```
Comment (sandris-) [inline src/main/java/io/codelex/flightplanner/weather/CacheKey.java]: are getters needed?

File: src/main/java/io/codelex/flightplanner/weather/WeatherGateway.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+
+@Component
+public class WeatherGateway {
+    private static final Logger LOG = LoggerFactory.getLogger(WeatherGateway.class);
```
Comment (sandris-) [inline src/main/java/io/codelex/flightplanner/weather/WeatherGateway.java]: is LOG used?

File: src/main/java/io/codelex/flightplanner/weather/WeatherGateway.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                JSONObject record = dayArr.getJSONObject(0);
+
+                forecast.setTemp_c(record.getJSONObject("day").getDouble("avgtemp_c"));
+                forecast.setWind_kph(record.getJSONObject("day").getDouble("maxwind_kph"));
+                forecast.setPrecip_mm(record.getJSONObject("day").getDouble("totalprecip_mm"));
+                forecast.setText(record.getJSONObject("day").getJSONObject("condition").getString("text"));
+                return Optional.of(forecast);
+
+            } catch (Exception e) {
+                e.printStackTrace();
```
Comment (sandris-) [inline src/main/java/io/codelex/flightplanner/weather/WeatherGateway.java]: this is a bad practice

File: src/test/java/io/codelex/flightplanner/ForecastCacheTest.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+class ForecastCacheTest {
```
Comment (sandris-) [inline src/test/java/io/codelex/flightplanner/ForecastCacheTest.java]: would be nice to check if caching works

File: src/main/java/io/codelex/flightplanner/api/TripWeather.java (hunk 1 above)
Comment (MixonZ) [reply src/main/java/io/codelex/flightplanner/api/TripWeather.java]: If there is a problem with weather service, weather will be null.
If flight requested, will return json: trip WITH weather or WITHOUT weather as a whole class.

Comment (sandris-) [reply src/main/java/io/codelex/flightplanner/api/TripWeather.java]: this does not answer to my question

File: src/main/java/io/codelex/flightplanner/api/Weather.java (hunk 2 above)
Comment (sandris-) [reply src/main/java/io/codelex/flightplanner/api/Weather.java]: not fixed

File: src/main/java/io/codelex/flightplanner/api/TripWeather.java (hunk 1 above)
Comment (sandris-) [reply src/main/java/io/codelex/flightplanner/api/TripWeather.java]: not fixed?

Comment (sandris-) [reply src/main/java/io/codelex/flightplanner/api/TripWeather.java]: is setter used somwhere?

**Pull request merged**